libraries/xcode.rb

```diff
@@ -23,11 +23,17 @@ def installed?(semantic_version)
 
         def highest_semantic_version(major_version)
           requirement = Gem::Dependency.new('iOS', "~> #{major_version}")
-          highest = available_list.select { |name, vers| requirement.match?(name, vers) }.max
-          if highest.nil?
-            Chef::Application.fatal!("iOS #{major_version} Simulator no longer available from Apple!")
+          if available_list.empty?
```

Comment:
This method has accumulated quite a few responsibilities:
- Creating a dependency object
- Checking the state of some (global?) list to see if it's populated yet and sleeping (retrying somehow?) if it isn't
- Creating a highest object by selecting the highest available simulator from the aformentioned list
- nil checking on the previous result so we can fail in a loud and obvious way
- and finally joining the highest object with spaces

I think we have a tortured object in here trying to get out.